tests/PHPStan/Analyser/data/bug-instanceof-static-vs-this-type-specifier.php

```diff
@@ -0,0 +1,37 @@
+<?php declare(strict_types=1);
+
+namespace BugInstanceofStaticVsThisTypeSpecifier;
+
+interface FooInterface
+{
+	/** @phpstan-assert-if-true null $v */
+	public static function isNull(?int $v): bool;
+
+	public static function foo(): int;
+}
+
+class FooBase
+{
+	public function bar(?int $v): void
+	{
+		if ($this instanceof FooInterface) {
+			if ($this->isNull($v)) \PHPStan\Testing\assertType('null', $v);
+			if ($this::isNull($v)) \PHPStan\Testing\assertType('null', $v);
+			if (static::isNull($v)) \PHPStan\Testing\assertType('null', $v);
+
+			if ($this::foo() === 5) \PHPStan\Testing\assertType('5', $this::foo());
```

Comment:
Ideally, this should work for any combination of `$this::`, `static::` and `$this::` (private methods might be an issue) for read/write. But for now it doesn't.